src/com/ms/silverking/cloud/zookeeper/ZooKeeperExtended.java

```diff
@@ -42,1024 +43,1081 @@
-        create(path, NumConversion.doubleToBytes(val));
+  }
+
+  public void waitForCompletion(List<? extends AsyncCallbackOp> ops) throws KeeperException {
+    for (AsyncCallbackOp op : ops) {
+      op.waitForCompletion();
     }
+  }
 
-    public double getDouble(String path) throws KeeperException {
-        try {
-            byte[] data;
-            Stat stat;
+  public VoidCallbackOp deleteAsync(String path) {
+    VoidCallbackOp op;
 
-            stat = new Stat();
-            data = getData(path, false, stat);
-            return NumConversion.bytesToDouble(data);
-        } catch (InterruptedException ie) {
-            throw new RuntimeException("panic");
-        }
+    op = newVoidCallbackOp();
+    delete(path, ANY_VERSION, this, op.getRequestUUID());
+    return op;
+  }
+
+  public void deleteRecursive(String path) throws KeeperException {
+    deleteChildrenRecursive(path);
+    delete(path);
+  }
+
+  private void deleteChildrenRecursive(String path) throws KeeperException {
+    List<String> children;
+    List<VoidCallbackOp> ops;
+
+    children = getChildren(path);
+    for (String child : children) {
+      deleteChildrenRecursive(path + "/" + child);
     }
-    
-    public StringCallbackOp createStringAsync(String path, String val) throws KeeperException {
-        return createAsync(path, val.getBytes());
+    ops = new ArrayList<>(children.size());
+    for (String child : children) {
+      ops.add(deleteAsync(path + "/" + child));
     }
+    waitForCompletion(ops);
+  }
 
-    public String createString(String path, String val, CreateMode createMode) throws KeeperException {
-        return create(path, val.getBytes(), createMode);
-    }
-    
-    public void createString(String path, String val) throws KeeperException {
-        create(path, val.getBytes());
+  public void createInt(String path, int val) throws KeeperException {
+    create(path, NumConversion.intToBytes(val));
+  }
+
+  public int getInt(String path) throws KeeperException {
+    return getInt(path, null);
+  }
+
+  public int getInt(String path, Watcher watcher) throws KeeperException {
+    try {
+      byte[] data;
+      Stat stat;
+
+      stat = new Stat();
+      data = getData(path, watcher, stat);
+      return NumConversion.bytesToInt(data);
+    } catch (InterruptedException ie) {
+      throw new RuntimeException("panic");
     }
-    
-    public void createInteger(String path, int v) throws KeeperException {
-        create(path, NumConversion.intToBytes(v));
+  }
+
+  public void createDouble(String path, double val) throws KeeperException {
+    create(path, NumConversion.doubleToBytes(val));
+  }
+
+  public double getDouble(String path) throws KeeperException {
+    try {
+      byte[] data;
+      Stat stat;
+
+      stat = new Stat();
+      data = getData(path, false, stat);
+      return NumConversion.bytesToDouble(data);
+    } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)